### PR TITLE
CHET-332: Meaningful errors

### PR DESCRIPTION
**Pull request by bpar476**
## Summary

* Add message parameter to error transition
* Store error message in migration context when transitioning
* Add some meaningful error messages in all error transitions
* Expose error message in migration summary API

### CODE REVIEW

**Comment by bpar476** [comment]
Lots of files changed because of many error messages 😢

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/DatabaseMigrationService.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public void abortMigration() throws InvalidMigrationStageError {
 
         logger.warn("Aborting running filesystem migration");
 
-        migrationService.error();
+        migrationService.error("file system migration was aborted");
```
Comment (nanux) [inline core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/DatabaseMigrationService.java:141]: /nit `File`

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/DatabaseRestoreStageTransitionCallback.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public void transitionToServiceNextStage() throws InvalidMigrationStageError {
     }
 
     @Override
-    public void transitionToServiceErrorStage() {
-        migrationService.error();
+    public void transitionToServiceErrorStage(String error) {
```
Comment (nanux) [inline core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/DatabaseRestoreStageTransitionCallback.java:49]: /nit - we can be probably consistent across all the `transitionToServiceErrorStage` and call the parameter `errorMessage` or just `error` (not `message`)

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/SsmPsqlDatabaseRestoreService.java [hunk 3]
```diff
@@ -58,9 +58,9 @@ public void restoreDatabase(DatabaseRestoreStageTransitionCallback restoreStageT
             consumer.handleCommandOutput(maxCommandRetries);
             restoreStageTransitionCallback.transitionToServiceNextStage();
         } catch (SuccessfulSSMCommandConsumer.UnsuccessfulSSMCommandInvocationException | SuccessfulSSMCommandConsumer.SSMCommandInvocationProcessingError e) {
-            restoreStageTransitionCallback.transitionToServiceErrorStage();
-            throw new DatabaseMigrationFailure("Unable to invoke database download command", e);
+            final String errorMessage = "Unable to invoke database download command";
```
Comment (nanux) [inline core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/SsmPsqlDatabaseRestoreService.java:61]: Couldn't this also mean that the database restore failed? E.g. the `pg_restore` script failed?

File: core/src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java [hunk 4]
```diff
@@ -109,7 +109,7 @@ public Boolean scheduleMigration() throws InvalidMigrationStageError
         boolean result = migrationRunner.runMigration(jobId, jobRunner);
 
         if (!result) {
-            migrationService.error();
+            migrationService.error("Error starting filesystem migration job. Check application logs for details");
```
Comment (nanux) [inline core/src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java:112]: I wonder if we shouldn't send people always to the `application logs` I would probably make the message part of the error page by default.

File: core/src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java [hunk 5]
```diff
@@ -171,7 +171,7 @@ public void abortMigration() throws InvalidMigrationStageError {
         fsUploader.abort();
         report.setStatus(FAILED);
 
-        migrationService.error();
+        migrationService.error("file system migration was aborted");
```
Comment (nanux) [inline core/src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java:174]: /nit `File`  - I think we should be consistent with the text if we are going to display it on he page.

File: core/src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java (hunk 4 above)
Comment (bpar476) [reply core/src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java:112]: Sounds good to me :)

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/SsmPsqlDatabaseRestoreService.java (hunk 3 above)
Comment (bpar476) [reply core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/SsmPsqlDatabaseRestoreService.java:61]: Correct! I'll update the string.

Comment (bpar476) [reply core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/SsmPsqlDatabaseRestoreService.java:61]: We could use the SSM command output to generate a more specific error message in future.

**Pull request merged**